### PR TITLE
EWF AMI Pipeline fixes

### PR DESCRIPTION
**Pull request by willwatters**

### CODE REVIEW

File: ansible/roles/heritage-frontend-base/vars/main.yml [hunk 1]
```diff
@@ -5,8 +5,19 @@ users:
...
-  - packages/oracle/oracle-instantclient11.2-sqlplus-11.2.0.2.0.x86_64.rpm
-  - application/ewf/frontend/ewf-frontend.tar.gz
+tar_files:
+  - ewf-frontend-app.tar.gz
+  - ewf-frontend-static.tar.gz 
+
+s3_file_app_static: /application/ewf/frontend/ewf-frontend-static.tar.gz
+s3_file_app: /application/ewf/frontend/ewf-frontend-app.tar.gz
+s3_files_app_static_name: ewf-frontend-static.tar.gz
+s3_files_app_name: ewf-frontend-app.tar.gz
```
Comment (MartyFox) [inline ansible/roles/heritage-frontend-base/vars/main.yml:15]: Should these be be in defaults.yml, you have defined them in host_vars/ as well which overrides vars and defaults in a role.
Lets makes sure we dont define variables in too many places or in different places between roles.

File: packer/build.pkr.hcl [hunk 2]
```diff
@@ -7,7 +7,8 @@ build {
     host_alias = "${var.ansible_host_alias}"
     playbook_file = "${var.playbook_file_path}"
     extra_arguments  = [
-      "-e", "aws_region=${var.aws_region}"
+      "-e", "aws_region=${var.aws_region}",
+      "-vv"
```
Comment (MartyFox) [inline packer/build.pkr.hcl:11]: can remove the -vv if we dont need verbose output anymore.

File: ansible/roles/heritage-frontend-base/vars/main.yml [hunk 3]
```diff
@@ -5,8 +5,19 @@ users:
...
+
+http_config:
+  - httpd/conf.d/ewf_perl.conf
+  - httpd/conf/ewf_startup.pl
+  - httpd/conf/httpd.conf
+
+user: ewf
+group: chlservices
```
Comment (MartyFox) [inline ansible/roles/heritage-frontend-base/vars/main.yml:23]: same as first comment, are these duplicates?

File: ansible/roles/heritage-frontend-base/tasks/main.yml [hunk 4]
```diff
@@ -22,110 +24,143 @@
...
+     name: "/tmp/{{ s3_files_oracle_sqlplus_rpm_name }}"
      state: present
-  with_items:
-    - oracle-instantclient11.2-basic-11.2.0.2.0.x86_64.rpm
-    - oracle-instantclient11.2-devel-11.2.0.2.0.x86_64.rpm
-    - oracle-instantclient11.2-sqlplus-11.2.0.2.0.x86_64.rpm
 
 - name: Ensure env variables are set
+  vars:
+    ansible_python_interpreter: /usr/bin/python2.6
```
[the remark this answers is not in the document]
Comment (willwatters) [reply ansible/roles/heritage-frontend-base/tasks/main.yml:94]: Yes it fails stating: "Aborting, target uses selinux but python bindings (libselinux-python) aren't installed!" and unable to find a selinux module for python3 on RHEL 6 so had it using /usr/bin/python2.6

File: ansible/roles/heritage-frontend-base/tasks/main.yml [hunk 5]
```diff
@@ -22,110 +24,143 @@
...
     path: "{{ user_home_path }}/.bash_profile"
     line: '{{ item }}'
-  with_items:
-   - export LD_LIBRARY_PATH=/usr/lib/oracle/11.2/client64/lib/
-   - export TNS_ADMIN=/usr/lib/oracle/11.2/client64/lib/
+  with_items: '{{ env_var }}'
 
 - name: Add config to ld.so.conf
+  vars:
+    ansible_python_interpreter: /usr/bin/python2.6
```
[the remark this answers is not in the document]
Comment (willwatters) [reply ansible/roles/heritage-frontend-base/tasks/main.yml:102]: Yes same as above for lineinfile command

File: ansible/roles/heritage-frontend-base/tasks/main.yml [hunk 6]
```diff
@@ -22,110 +24,143 @@
...
-    - /usr/local/lib
+  with_items: '{{ ld_so_conf }}'
 
 - name: Run ldconfig
   command: /sbin/ldconfig
 
 - name: Remove any exisitng Perl binaries
   file:
-    path: "{{ item }}"
+    path: '{{ item }}'
```
[the remark this answers is not in the document]
Comment (willwatters) [reply ansible/roles/heritage-frontend-base/tasks/main.yml:113]: Changed to: path: "{{ item }}"

File: ansible/roles/heritage-frontend-base/tasks/main.yml [hunk 7]
```diff
@@ -22,110 +24,143 @@
...
 
 - name: Remove installation files
   file:
     path: "{{ item.path }}"
     state: absent
   with_items: "{{ files_to_delete.files }}"
+
+- name: Remove any exisitng httpd.conf
+  file:
+    path: '{{ item }}'
```
[the remark this answers is not in the document]
Comment (willwatters) [reply ansible/roles/heritage-frontend-base/tasks/main.yml:152]: changed to path: "{{ item }}"

File: packer/build.pkr.hcl (hunk 2 above)
Comment (willwatters) [reply packer/build.pkr.hcl:11]: Now removed:

 extra_arguments  = [
      "-e", "aws_region=${var.aws_region}"
    ]

File: ansible/host_vars/ewf-frontend.yaml [hunk 8]
```diff
@@ -1,15 +1,23 @@
...
-   version: "1.7.36"
+
+tar_files:
+  - ewf-frontend-app.tar.gz
+  - ewf-frontend-static.tar.gz 
+
+s3_file_app_static: /application/ewf/frontend/ewf-frontend-static.tar.gz
+s3_file_app: /application/ewf/frontend/ewf-frontend-app.tar.gz
+s3_files_app_static_name: ewf-frontend-static.tar.gz
+s3_files_app_name: ewf-frontend-app.tar.gz
```
[the remark this answers is not in the document]
Comment (willwatters) [reply ansible/host_vars/ewf-frontend.yaml:15]: Changed to:

s3_file_app_static: /application/ewf/frontend/ewf-frontend-static.tar.gz
s3_file_app: /application/ewf/frontend/ewf-frontend-app.tar.gz
s3_files_app_static_name: {{ s3_file_app_static | basename }}
s3_files_app_name: {{ s3_file_app | basename }}

File: ansible/roles/heritage-frontend-base/vars/main.yml (hunk 1 above)
Comment (willwatters) [reply ansible/roles/heritage-frontend-base/vars/main.yml:15]: The vars have been removed from the role var to defaults and host_vars removed.

File: ansible/host_vars/ewf-frontend.yaml [hunk 9]
```diff
@@ -1,15 +1,23 @@
...
+
+http_config:
+  - httpd/conf.d/ewf_perl.conf
+  - httpd/conf/ewf_startup.pl
+  - httpd/conf/httpd.conf
+
+user: ewf
+group: chlservices
```
[the remark this answers is not in the document]
Comment (willwatters) [reply ansible/host_vars/ewf-frontend.yaml]: Duplicate vars removed and replaced with the following in role vars, with code updated to use new vars:

users:
 - service_user
 
user_groups:
  - chlservices

File: ansible/roles/heritage-frontend-base/vars/main.yml (hunk 3 above)
Comment (willwatters) [reply ansible/roles/heritage-frontend-base/vars/main.yml]: Duplicate vars removed as stated above.

**Review by MartyFox — APPROVED** [review]

**Pull request merged**